tensorflow/python/distribute/cluster_resolver/sagemaker_cluster_resolver.py

```diff
@@ -0,0 +1,208 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Implementation of Cluster Resolvers for SageMaker Environment."""
+
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import json
+import os
+
+from tensorflow.python.distribute.cluster_resolver.cluster_resolver import ClusterResolver
+from tensorflow.python.training.server_lib import ClusterSpec
+from tensorflow.python.util.tf_export import tf_export
+
+#List of envs: https://github.com/aws/sagemaker-training-toolkit/blob/master/ENVIRONMENT_VARIABLES.md
+#Only support Multi-Worker Mirrored Strategy
+
+_SESSION_MASTER_KEY = 'session_master'
+_RPC_LAYER_KEY = 'rpc_layer'
+_TASK_KEY = 'task'
+_CLUSTER_KEY = 'cluster'
+_WORKER_KEY = 'worker'
+_INDEX_KEY = 'index'
+_TYPE_KEY = 'type'
+
+_SM_CURRENT_HOST = "SM_CURRENT_HOST"
+_SM_HOSTS = "SM_HOSTS"
+
+def format_master_url(master, rpc_layer=None):
+  if rpc_layer:
+    return '%s://%s' % (rpc_layer, master)
+  else:
+    return master
+
+
+def _load_tf_config(port):
+  #Create a tf_config from SM Variables
+  assert all([x in os.environ for x in [_SM_CURRENT_HOST, _SM_HOSTS]]),\
```

Comment:
According to the Google Python style guide, no blackslash continuations here. Consider using implicit line joining inside parentheses instead.